apistructs/pipeline_yml.go

```diff
@@ -111,6 +118,36 @@ type SnippetConfig struct {
 	Labels map[string]string `json:"labels,omitempty" yaml:"labels,omitempty"` // 额外标签
 }
 
+type SnippetLabel struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type SnippetLabels []SnippetLabel
+
+func (p SnippetLabels) Len() int           { return len(p) }
+func (p SnippetLabels) Less(i, j int) bool { return p[i].Key > p[j].Key }
+func (p SnippetLabels) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+func (snippetConfig *SnippetConfig) ToString() string {
+	if snippetConfig == nil {
+		return ""
+	}
+	var labels string
+	if len(snippetConfig.Labels) > 0 {
+		var snippetLabels SnippetLabels
+		for k, v := range snippetConfig.Labels {
+			snippetLabels = append(snippetLabels, SnippetLabel{
+				Key:   k,
+				Value: v,
+			})
+		}
+		sort.Sort(snippetLabels)
+		labels = jsonparse.JsonOneLine(snippetLabels)
+	}
+	return fmt.Sprintf("{\"source\": %s, \"name\": %s, \"labels\": %s}", snippetConfig.Source, snippetConfig.Name, labels)
```

Comment:
Values of labels will have quote problems.
You can define a anonymous struct and json marshal it.